newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -135,223 +67,67 @@ type Props = {|
-                });
-              };
-
-              return (
-                <div
-                  style={{
-                    ...styles.container,
-                    flexGrow: props.fullWidth ? 1 : undefined,
-                  }}
-                >
-                  {renderTextField({
-                    disabled: props.disabled,
-                    label: props.floatingLabelText,
-                    id: props.id,
-
-                    // Error handling:
-                    error: !!props.errorText,
-                    helperText: props.errorText || helperText,
-
-                    // Display:
-                    InputLabelProps: getLabelProps({ shrink: true }),
-
-                    // Events:
-                    InputProps: {
-                      // We wrap the onBlur/onFocus as we're a "semi controlled" field
-                      onBlur: wrappedOnBlur,
-                      onFocus: onFocus,
-                    },
-
-                    // Props for the input field from downshift:
-                    inputProps,
-
-                    // Style:
-                    style: props.textFieldStyle,
-                    fullWidth: props.fullWidth,
-                    ...computeTextFieldStyleProps(props),
-
-                    inputRef: this._input,
-                  })}
-                  <Popper
-                    open={isOpen}
-                    anchorEl={this._input.current}
-                    style={styles.menuPopper}
-                  >
-                    <div
-                      {...(isOpen
-                        ? getMenuProps({}, { suppressRefError: true })
-                        : {})}
-                    >
-                      {isOpen ? (
-                        <Paper
-                          style={{
-                            ...styles.menuPaper,
-                            width: this._input.current
-                              ? this._input.current.clientWidth
-                              : undefined,
-                          }}
-                          square
-                        >
-                          {filterDataSource(props.dataSource, inputValue).map(
-                            (item, index) =>
-                              renderItem({
-                                item,
-                                index,
-                                menuItemProps: getItemProps({ item, index }),
-                                highlightedIndex,
-                                selected: item === selectedItem,
-                              })
-                          )}
-                        </Paper>
-                      ) : null}
-                    </div>
-                  </Popper>
-                </div>
-              );
-            }}
-          </Downshift>
-        )}
-      </I18n>
-    );
-  }
-}
+          <MarkdownText source={props.helperMarkdownText} />
+          ) : null;
+
+    const classes = styles();
```

Comment:
it's very surprising to have a function called `styles` => find a name with a verb :)